src/BIP44CoinTypeNode.test.ts

```diff
@@ -89,80 +89,99 @@ describe('BIP44CoinTypeNode', () => {
-        [defaultBip39NodeToken, BIP44PurposeNodeToken, `bip32:60'`, `bip32:0'`],
-      ].forEach((invalidDerivationPath) => {
-        expect(
-          () => new BIP44CoinTypeNode(invalidDerivationPath as any),
-        ).toThrow(
-          `Invalid depth: Coin type nodes must be of depth ${BIP_44_COIN_TYPE_DEPTH}. Received: "${
-            invalidDerivationPath.length - 1
-          }"`,
-        );
-      });
+    it('throws if derivation path has invalid depth', async () => {
+      await expect(() =>
+        BIP44CoinTypeNode.create([
+          defaultBip39NodeToken,
+          BIP44PurposeNodeToken,
+        ] as any),
+      ).rejects.toThrow(
+        `Invalid depth: Coin type nodes must be of depth ${BIP_44_COIN_TYPE_DEPTH}. Received: "1"`,
+      );
+
+      await expect(() =>
+        BIP44CoinTypeNode.create([
+          defaultBip39NodeToken,
+          BIP44PurposeNodeToken,
+          `bip32:60'`,
+          `bip32:0'`,
+        ] as any),
+      ).rejects.toThrow(
+        `Invalid depth: Coin type nodes must be of depth ${BIP_44_COIN_TYPE_DEPTH}. Received: "3"`,
+      );
     });
 
-    it('throws if node has invalid depth', () => {
+    it('throws if node has invalid depth', async () => {
       const arbitraryCoinType = 78;
 
-      [
-        { key: 'foo', depth: 1 },
-        { key: 'foo', depth: 3 },
-      ].forEach((invalidNode) => {
-        expect(
-          () => new BIP44CoinTypeNode(invalidNode as any, arbitraryCoinType),
-        ).toThrow(
-          `Invalid depth: Coin type nodes must be of depth ${BIP_44_COIN_TYPE_DEPTH}. Received: "${invalidNode.depth}"`,
-        );
-      });
+      await expect(() =>
+        BIP44CoinTypeNode.create(
+          { key: 'foo', depth: 1 } as any,
+          arbitraryCoinType,
+        ),
+      ).rejects.toThrow(
+        `Invalid depth: Coin type nodes must be of depth ${BIP_44_COIN_TYPE_DEPTH}. Received: "1"`,
+      );
+
+      await expect(() =>
+        BIP44CoinTypeNode.create(
+          { key: 'foo', depth: 3 } as any,
+          arbitraryCoinType,
+        ),
+      ).rejects.toThrow(
+        `Invalid depth: Coin type nodes must be of depth ${BIP_44_COIN_TYPE_DEPTH}. Received: "3"`,
+      );
     });
 
-    it('throws if node has invalid key', () => {
+    it('throws if node has invalid key', async () => {
       const arbitraryCoinType = 78;
-
-      [
+      const inputs = [
         { key: 'foo', depth: 2 },
         { key: 1, depth: 2 },
         { key: Buffer.allocUnsafe(64).fill(1).toString('hex'), depth: 2 },
         { key: Buffer.allocUnsafe(63).fill(1).toString('base64'), depth: 2 },
         { key: Buffer.alloc(64).toString('base64'), depth: 2 },
-      ].forEach((invalidNode) => {
-        expect(
-          () => new BIP44CoinTypeNode(invalidNode as any, arbitraryCoinType),
-        ).toThrow('Invalid parent key: Must be a non-zero 64-byte key.');
-      });
+      ];
+
+      for (const input of inputs) {
```

Comment:
Why are we replacing `forEach` with `for .. of ..` here?

Reply:
Can't use `await` with `forEach`.